### DISCUSSION TITLE
Using current directory as cwd when opening a file

### DISCUSSION THREAD

**Discussion by kristijanhusak** [jesseduffield/lazygit#3160] [Q&A]
Hi,

Recently I started working with monorepo which has a lot of sub-packages.
When I'm in the subpackage folder and I open lazygit + press `e` to edit a file, it opens up Neovim with CWD being the root directory (git root). Is there a way to edit the file with cwd being the same as the one from where I opened lazygit?

For example, assuming this structure and assuming Vim (Or Neovim) is the editor:

```
myrepo
  .git
  packages/
  		frontend/
    		readme_frontend.md
  		backend/
    		readme_backend.md
```

doing this:
1. `cd packages/frontend`
2. `lazygit`
3. Inside lazygit, press `e` to edit `readme_frontend.md`

In Vim, checking the cwd with `:pwd` outputs `/path/to/myrepo`.

And I would like it to actually use the real cwd, where the `:pwd` would output `/path/to/myrepo/packages/frontend`

I mostly need this because I have some local configuration scoped to the specific package (local vimrc), and when I open the Neovim from lazygit, it does not pick that up. Also, the general usage with the cwd being git root gives me a lot of results in fuzzy finder, etc.

I tried using this as an edit command but it didn't work:

```
os:
  edit: 'nvim --cmd "cd `pwd`" -- {{filename}}'
```

**Comment by mark2185** [comment]
> I tried using this as an edit command but it didn't work:
> 
> ```
> os:
>   edit: 'nvim --cmd "cd `pwd`" -- {{filename}}'
> ```

Isn't `pwd` the wrong argument, though? I'm guessing it returns the root of the clone, but you'd actually want `dirname {{ filename }}` or something of sorts?

**Reply by kristijanhusak** [reply]
Yeah `pwd` is wrong, it returns the git root. `dirname {{filename}}` would work in this specific example I gave, but I'm actually having more nesting inside the subpackage. Ideally I would be able to use same cwd from where the lazygit was opened.

**Reply by mark2185** [reply]
`"cd $(git -C $(dirname {{ filename }}) rev-parse --show-toplevel)"`?

To be honest, it looks like there should be an easier way.

**Reply by kristijanhusak** [reply]
You mean to use that command instead of "cd `pwd`"  like this:

```
nvim --cmd "cd `$(git -C $(dirname {{ filename }}) rev-parse --show-toplevel)`" -- {{filename}}
```

It actually makes my cwd go up some levels. Also, this is passed down to Neovim `cmd` which cannot interpolate bash with 
`$()` so I wrapped it in backticks to be executed as part of bash command. Anyway, it does not work unfortunately.

>  To be honest, it looks like there should be an easier way.

Yeah I'm hoping that there is. I wasn't able to find any though.

**Reply by mark2185** [reply]
Well another workaround that comes to mind is to make the invocation of `lazygit` a shell function which sets a variable `USING_LAZYGIT=1` and then to check in your `nvimrc` for the existance of said variable and to change the current directory via the `git -C` incantation.

**Comment by Raaghav-Arya** [comment]
Was facing a similar issue (automatic cwd change). Realised the issue was with project.nvim. Here is the thread:
https://github.com/LazyVim/LazyVim/discussions/6857
